src/documentation.lisp

```diff
@@ -29,61 +29,39 @@
             (do-all-external-symbols (symbol) (handle-symbol symbol))
             (do-all-symbols (symbol) (handle-symbol symbol))))))
 
-(defun apropos-list (string &optional package external-only)
-  (let (symbols)
-    (map-apropos-symbols
-     (lambda (symbol)
-       (pushnew symbol symbols :test #'eq))
-     string package external-only)
-    symbols))
+(defun apropos/regexp-test (pattern str)
+  ((oget pattern "test")  str))
 
-(defun apropos (string &optional package external-only)
-  (map-apropos-symbols
-   (lambda (symbol)
-     (format t "~S" symbol)
-     (when (boundp symbol)
-       (format t " (bound)"))
-     (when (fboundp symbol)
-       (format t " (fbound)"))
-     (terpri))
-   (string string) package external-only))
-
-;;; DESCRIBE
+(defun apropos-list (string &optional package externals-only)
```

Comment:
Does the standard allows using regular expressions for apropos?

> These functions search for interned symbols whose names contain the substring string.

It's probably a good extension. But should we not built-in too much in it?  What if we define

`(defvar *apropos-comparison-funtion* ..)`

so we can easily change it to use regexs or substrings?

Reply:
Yes, its good idea.
And it requires good implementation.
In `ToDo`